controllers/cloud.redhat.com/providers/providers.go

```diff
@@ -184,13 +184,13 @@ type ObjectMap map[rc.ResourceIdent]client.Object
 
 // CachedMakeComponent is a generalised function that, given a ClowdObject will make the given service,
 // deployment and PVC, based on the makeFn that is passed in.
-func CachedMakeComponent(cache *rc.ObjectCache, objList []rc.ResourceIdent, o obj.ClowdObject, suffix string, fn makeFnCache, usePVC bool, nodePort bool) error {
+func CachedMakeComponent(prov RootProvider, objList []rc.ResourceIdent, o obj.ClowdObject, suffix string, fn makeFnCache, usePVC bool) error {
 	nn := GetNamespacedName(o, suffix)
 
 	makeFnMap := make(map[rc.ResourceIdent]client.Object)
 
 	for _, v := range objList {
-		obj, err := createResource(cache, v, nn)
+		obj, err := createResource(prov.GetCache(), v, nn)
```

Comment:
Do we have any concern that an incorrectly implemented provider might not have this method? Like, do we need to check for it first or something before calling it? I was worried about a nil pointer.